test/testbufferoverrun.cpp

```diff
@@ -5597,15 +5607,13 @@ class TestBufferOverrun : public TestFixture {
               "    char b;\n"
               "};\n"
               "void f() {\n"
-              "    X s;\n"
-              "    int* y = &s.a;\n"
+              "    const X s;\n"
+              "    const int* y = &s.a;\n"
               "    (void)y[0];\n"
               "    (void)y[1];\n"
               "    (void)y[2];\n"
               "}\n");
-        ASSERT_EQUALS("[test.cpp:7] -> [test.cpp:9]: (error) The address of local variable 'a' is accessed at non-zero index.\n"
```

Comment:
It's unfortunate that it does not warn anymore but I feel the implementation and message was sloppy:
 * Variable 'a' is not a local variable. I guess it should have warned about 's.a' instead.
 * It does not matter if it is a local or global variable what is important is that it's a plain int variable. Therefore it feels wrong to use the lifetime values in the checker wouldn't it be better to rely on tokvalue?

Reply:
about these comments: I created ticket https://trac.cppcheck.net/ticket/12428 so we can act on those later when we have time.